osu.Game.Rulesets.Mania/UI/Column.cs

```diff
@@ -31,109 +26,60 @@ public class Column : ScrollingPlayfield, IKeyBindingHandler<ManiaAction>, IHasA
 
         public readonly Bindable<ManiaAction> Action = new Bindable<ManiaAction>();
 
-        private readonly ColumnBackground background;
-        private readonly ColumnKeyArea keyArea;
-        private readonly ColumnHitObjectArea hitObjectArea;
+        protected abstract Container HitTargetContainer { get; }
+        protected abstract Drawable KeyArea { get; }
+        protected abstract Container ExplosionContainer { get; }
+        protected internal abstract Container TopLevelContainer { get; }
 
-        internal readonly Container TopLevelContainer;
-        private readonly Container explosionContainer;
-
-        public Column(int index)
+        protected Column(int index)
         {
             Index = index;
 
             RelativeSizeAxes = Axes.Y;
             Width = COLUMN_WIDTH;
+        }
 
-            background = new ColumnBackground { RelativeSizeAxes = Axes.Both };
+        public static Column CreateColumnFromType(ColumnType type, int index)
```

Comment:
Don't like this being `static` and part of `Column`. Should be virtual (or abstract) and part of `ManiaStage` instead.